BMM.Core/Implementations/Player/PlaybackHistoryService.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Akavache;
+using BMM.Api.Abstraction;
+using BMM.Api.Implementation.Models;
+using BMM.Core.Helpers;
+using BMM.Core.Implementations.Persistence.Interfaces;
+using BMM.Core.Implementations.Player.Interfaces;
+using BMM.Core.Models.PlaybackHistory;
+using BMM.Core.Models.PlaybackHistory.Interfaces;
+
+namespace BMM.Core.Implementations.Player
+{
+    public class PlaybackHistoryService : IPlaybackHistoryService
+    {
+        public const int MaxEntries = 100;
+        private readonly IBlobCacheWrapper _blobCacheWrapper;
+
+        public PlaybackHistoryService(
+            IBlobCacheWrapper blobCacheWrapper)
+        {
+            _blobCacheWrapper = blobCacheWrapper;
+        }
+
+        public async void AddPlayedTrack(IMediaTrack mediaTrack)
```

Comment:
Isn't async void dangerous and should be prevented if possible?
Also, do we need to worry about concurrency? e.g. quickly clicking multiple tracks.

Reply:
Added semaphore